osu.Game/Overlays/Mods/BpmDisplay.cs

```diff
@@ -0,0 +1,209 @@
+                                            Shear = new Vector2(-ShearedOverlayContainer.SHEAR, 0),
+                                            Text = "Average BPM",
+                                            Font = OsuFont.Default.With(size: 17, weight: FontWeight.SemiBold)
+                                        }
+                                    }
+                                },
+                                new Container
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Shear = new Vector2(-ShearedOverlayContainer.SHEAR, 0),
+                                    AutoSizeAxes = Axes.Both,
+                                    Children = new Drawable[]
+                                    {
+                                        counter = new BpmCounter
+                                        {
+                                            Current = { BindTarget = current }
+                                        },
+                                        dashText = new OsuSpriteText
+                                        {
+                                            Anchor = Anchor.Centre,
+                                            Origin = Anchor.Centre,
+                                            Font = OsuFont.Default.With(size: 17, weight: FontWeight.SemiBold),
+                                            Text = "-"
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            contentBackground.Colour = colourProvider.Background4;
+        }
+
+        protected override void LoadComplete()
+        {
+            mods.BindValueChanged(m =>
+            {
+                settingChangeTracker?.Dispose();
+
+                refresh();
+
+                settingChangeTracker = new ModSettingChangeTracker(m.NewValue);
+                settingChangeTracker.SettingChanged += _ => refresh();
+            }, true);
+            working.ValueChanged += _ => refresh();
+        }
+
+        /// <summary>
+        /// Refreshes counter and background color.
+        /// </summary>
+        private void refresh()
+        {
+            var beatmap = working.Value.Beatmap;
+
+            if (beatmap == null)
+                return;
+
+            double rate = 1;
+            foreach (var mod in mods.Value.OfType<IApplicableToRate>())
+                rate = mod.ApplyToRate(0, rate);
+
+            double mostCommonBPM = Math.Round(Math.Round(60000 / beatmap.GetMostCommonBeatLength()) * rate);
+
+            if (double.IsNormal(mostCommonBPM))
```

Comment:
What is this `IsNormal` check, even? Is this supposed to catch NaNs? When can this even occur? A comment below is saying something about "no map selected" - maybe the way there is to, you know, not divide by 0 in the first place?